Loosened to `dist <= stop_thresh` to converge in on 1D constant data (#28951)
scikit-learn/scikit-learn: doc/whats_new/v1.5.rst

Changed models
--------------

> - |Efficiency| The `clustering.MeanShift` class has now improved computational speed as it properly converges for constant data.
>   :pr:`28951` by :user:`Akihiro Kuno <akikuno>`.

Reviewer: ```suggestion
- |Fix| The :class:`cluster.MeanShift` class now properly converges for constant data.
  :pr:`28951` by :user:`Akihiro Kuno <akikuno>`.
```

Reviewer 2: Once this suggestion is applied, this entry of the changelog should subsequently be moved to the `sklearn.cluster` section of this page (if it already exists and create it if not).

> 

Reviewer: I would consider it a bug and move that in the `cluster` section of the changelog.
